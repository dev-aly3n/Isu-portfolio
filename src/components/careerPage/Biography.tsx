// libs
import {AnimateSharedLayout} from 'framer-motion';
// components
import BioChapter from "./BioChapter";
// hooks
// store
import {biochapters} from '../../store/allData';
// utils & animation



const Biography: React.FC = () => {
  return (
    <div className="flex flex-col relative w-full h-full py-20 px-60">
        {biochapters[0] && biochapters.map((chap,index)=>{
            return <BioChapter chap={chap} firstChap={index === 0 ? true : false} />
        })}
    </div>
  );
};

export default Biography;
