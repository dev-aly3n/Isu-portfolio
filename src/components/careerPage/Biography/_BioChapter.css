@layer components {
  .bio-chapter-container {
    @apply flex flex-col justify-center items-center w-10/12 sm:w-full my-3;
    > h3 {
      @apply no-select-a-tag text-2xl font-bold self-start pl-5 cursor-pointer w-full;
    }
    > div {
      @apply w-full h-px bg-white my-2;
    }
    > p {
      @apply text-base first-letter:text-primary-200 first-letter:font-semibold first-letter:text-5xl overflow-hidden;
    }
  }
}
