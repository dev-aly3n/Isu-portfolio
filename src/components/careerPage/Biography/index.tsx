// libs
import {motion} from 'framer-motion';
// components
import BioChapter from "./BioChapter";
// hooks
// store
import {biochapters} from '../../../store/allData';
import { RefObject } from 'react';
// utils & animation

interface props {
  bioRef:RefObject<HTMLDivElement>;
}

const Biography: React.FC<props> = ({bioRef}) => {
  return (
    <div ref={bioRef}>
      <motion.h2 layout>Biography</motion.h2>
        {biochapters[0] && biochapters.map((chap,index)=>{
            return <BioChapter key={chap.title} chap={chap} firstChap={index === 0 ? true : false} />
        })}
    </div>
  );
};

export default Biography;
