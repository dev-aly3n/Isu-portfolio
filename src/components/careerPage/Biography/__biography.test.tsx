import { render, screen } from "@testing-library/react";
import React, { useRef } from "react";
import { RefObject } from "react";
import Biography from ".";

const ref = React.createRef<HTMLDivElement>();


describe("career/Biography", () => {
  it("snap", () => {
    const { container } = render(<Biography bioRef={ref} />);
    expect(container).toMatchSnapshot();
  });
});
