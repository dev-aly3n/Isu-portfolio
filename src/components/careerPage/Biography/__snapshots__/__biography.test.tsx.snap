// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`career/Biography snap 1`] = `
<div>
  <div>
    <h2>
      Biography
    </h2>
    <div
      class="bio-chapter-container"
      style="opacity: 0;"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-right fa-w-8 rotate-90 duration-500"
          data-icon="angle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 256 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            fill="currentColor"
          />
        </svg>
         
        Chapter One: The Beginning
      </h3>
      <div />
      <p
        class=" "
      >
        Aly Mohamadi was born on 9 February 1998 in Iran. There is nothing fancy about his family, just a regular one with all problems in the world that a family could has. Anyway he raised and on the very old day at school he watched the movie of 'October Sky' that had been made base on a real story of a book, Rocket Boys. Story is about the life of its author Hoomer Hickam. He was born in very fucked up town that all who was born there, would be miner eventually. But Hoomer after all didn't give up and tried to reach to his dream, being a rocket boy and working in NASA.
      </p>
    </div>
    <div
      class="bio-chapter-container"
      style="opacity: 0;"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-right fa-w-8 rotate-0 duration-500"
          data-icon="angle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 256 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            fill="currentColor"
          />
        </svg>
         
        Chapter Two: The Wish and The Failure
      </h3>
      <div />
      <p
        class=" h-0"
      >
        So it's all and this is what happened to very young Aly. He started dreaming about it and he tried to be a rocket boy but in Iran?! His place was more fucked up than Hoomer's town. He went to university and attempted to start with cheaper stuff so made some water rockets and tried to improve them but you know what? He couldn't find a hand-pump that could survive in 20 bar pressure (about 20 times of atmosphere). Yes I know you can't believe it. But it's true. Why, because this country of mullahs has been boycotted. So Aly hadn't the stuff that his friend Julian bought from a store besides of his home in Germany. Anyway he tried even cheaper, Aly created a magazine about Aerospace and named it ASPAD. He made a website with Joomla as he worked with it since he was 14 and a blogger who wrote about rockets and water rockets. He worked on it days and nights. He even set an interview with Hoomer Hickam that eventually Hoomer answered his question very kindly. The website and magazine was so interesting but you know free volunteering works in a country that falling to bottom of poverty is not working. So it's the time for new chapter.
      </p>
    </div>
    <div
      class="bio-chapter-container"
      style="opacity: 0;"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-right fa-w-8 rotate-0 duration-500"
          data-icon="angle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 256 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            fill="currentColor"
          />
        </svg>
         
        Chapter Three: Changing The Career
      </h3>
      <div />
      <p
        class=" h-0"
      >
        He started thinking about coding. He searched for the job offers. Aly needed inspiring to start it. Finally he found it. Inspired by growing of this industry and how creativity that he could put on it. So he started to learn. He learned html css and js from w3school. I know I know such a nice place for learning nothing. He wasted one year on w3school besides of his study of aerospace. But after a year he found out that there is better materials and started to learn from an iranian teacher and please let me bang my head against the wall. That was horrible. But one day he searched how to build a nav bar css js and found a guy whose name was Dev Ed on Youtube. Aly loved his course and his ability to english speaking. In result he searched about this guy and found his courses about js. So started learning again. Working on his code for days and nights with ultimate creativity that he had. How ever he had a lots of problem and left the coding for days but he came back to continue and completed his projects.
      </p>
    </div>
    <div
      class="bio-chapter-container"
      style="opacity: 0;"
    >
      <h3>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-right fa-w-8 rotate-0 duration-500"
          data-icon="angle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 256 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            fill="currentColor"
          />
        </svg>
         
        Chapter Four: To be continued
      </h3>
      <div />
      <p
        class=" h-0"
      >
        After all Aly completed 8 creative projects with different technologies that he learnt and he is still learning. But the most important thing that will happen in this chapter is finding a job. Yes Aly will find a nice job at this chapter.  He thinks that he is ready to work with a nice team.
      </p>
    </div>
  </div>
</div>
`;
